compiler/plc_ast/src/ast.rs

```diff
@@ -1047,44 +1051,62 @@ impl AstStatement {
         // TODO: figure out a better name for this...
         match self {
             AstStatement::Literal { kind, .. } => kind.is_cast_prefix_eligible(),
-            AstStatement::Reference { .. } => true,
+            AstStatement::Identifier { .. } => true,
             _ => false,
         }
     }
 
-    /// Returns true if the current statement is a reference
-    pub fn is_reference(&self) -> bool {
-        matches!(self, AstStatement::Reference { .. })
+    /// Returns true if the current statement is a flat reference (e.g. `a`)
+    pub fn is_flat_reference(&self) -> bool {
+        matches!(self, AstStatement::Identifier { .. }) || {
+            if let AstStatement::ReferenceExpr {
+                access: ReferenceAccess::Member(reference),
+                base: None,
+                ..
+            } = self
+            {
+                matches!(reference.as_ref(), AstStatement::Identifier { .. })
```

Comment:
I reconsidered this, it is fine. It checks wheter the referenceExpression points to a Identifier, rather than for example a Literal.
When we discusse dit, I thought there is an OR